src/cryptography/hazmat/primitives/padding.py

```diff
@@ -122,3 +128,26 @@ def finalize(self):
         res = self._buffer[:-pad_size]
         self._buffer = None
         return res
+
+
+class ANSIX923(PKCS7):
```

Comment:
cryptography as a project has a general policy against subclassing. In this case I'd recommend creating a separate class and then factor out the block size checks from the `PKCS7` class so they can be called from both PKCS7 and the ANSIX923 classes.